pkg/notification/db_changes.sql

```diff
@@ -45,9 +45,5 @@ CREATE INDEX `notification_recipients_ix1` ON `notification_recipients` (`recipi
 
 -- changes for added notifications on oc shares
 
-ALTER TABLE oc_share ADD notify_uploads BOOL DEFAULT false;
+ALTER TABLE oc_share ADD notify_uploads INTEGER DEFAULT 0 NOT NULL;
```

Comment:
Why an Integer?

Reply:
Because I suggested - a bit late indeed - to have a more future-proof column for this, imaging that in the future we want to enable notifications for different types of events (and even the name is misleading then). But I realize after talking to @javfg that we have to review the whole stack, CS3APIs included. Sorry for not having picked up this when we reviewed the CS3API change...

At this stage, for consistency this is to be reverted to `BOOL`, but I'd still like to check what it entails to change it properly across the stack. Remember that once we apply this change in prod, we create a column for 490K entries - which itself takes quite a bit of time and takes an exclusive lock on the table! - and it would be a pity to have to change that or to need to add yet another column afterwards.

Reply:
This is why we should normalize this table :)

Reply:
I agree!